src/kafka-file/kafkaFileClient.ts

```diff
@@ -0,0 +1,118 @@
+import * as vscode from "vscode";
+import { ConsumerCollection, ConsumerCollectionChangedEvent, ConsumerLaunchState } from "../client/consumer";
+import { ProducerCollection, ProducerCollectionChangedEvent, ProducerLaunchState } from "../client/producer";
+import { ClusterSettings } from "../settings/clusters";
+
+import { getLanguageModelCache, LanguageModelCache } from './languageModelCache';
+import { KafkaFileDocument } from "./languageservice/parser/kafkaFileParser";
+import { ConsumerLaunchStateProvider, getLanguageService, LanguageService, ProducerLaunchStateProvider, SelectedClusterProvider } from "./languageservice/kafkaFileLanguageService";
+import { runSafeAsync } from "./runner";
+
+class KafkaFileCodeLensProvider implements vscode.CodeLensProvider, vscode.Disposable {
+    private disposables: vscode.Disposable[] = [];
+    private _onDidChangeCodeLenses: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
+    public readonly onDidChangeCodeLenses: vscode.Event<void> = this._onDidChangeCodeLenses.event;
+
+    constructor(
+        private clusterSettings: ClusterSettings,
+        private producerCollection: ProducerCollection,
+        private consumerCollection: ConsumerCollection,
+        private kafkaFileDocuments: LanguageModelCache<KafkaFileDocument>,
+        private languageService: LanguageService
+    ) {
+        // Refresh the code lenses when:
+        // 1. a consumer is started / stopped to refresh the status of each declared CONSUMER
+        this.disposables.push(this.consumerCollection.onDidChangeCollection((e: ConsumerCollectionChangedEvent) => {
+            this._onDidChangeCodeLenses.fire();
+        }));
+        // 2. a producer is started / stopped to refresh the status of each declared PRODUCER
+        this.disposables.push(this.producerCollection.onDidChangeCollection((e: ProducerCollectionChangedEvent) => {
+            this._onDidChangeCodeLenses.fire();
+        }));
+        // 3. a cluster is selected
+        this.clusterSettings.onDidChangeSelected((e) => {
+            this._onDidChangeCodeLenses.fire();
+        });
+    }
+
+    provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken): vscode.ProviderResult<vscode.CodeLens[]> {
+        return runSafeAsync(async () => {
+            const kafkaFileDocument = this.kafkaFileDocuments.get(document);
+            return this.languageService.getCodeLenses(document, kafkaFileDocument);
+        }, [], `Error while computing code lenses for ${document.uri}`, token);
+    }
+
+    dispose() {
+        this.disposables.forEach(d => d.dispose());
+    }
+}

```

Comment:
I'd move to its own file

Reply:
This class is just a wrapper which consumes language service, so I would like to keep it in this client file. I restructured the file, please tell me if it's better.